### ISSUE TITLE
🐛 Biome encountered an unexpected error

### ISSUE THREAD

**Issue by monkfromearth** [biomejs/biome#2679]
### Environment information

```block
CLI:
  Version:                      1.7.2
  Color support:                true

Platform:
  CPU Architecture:             x86_64
  OS:                           linux

Environment:
  BIOME_LOG_DIR:                unset
  NO_COLOR:                     unset
  TERM:                         "xterm-256color"
  JS_RUNTIME_VERSION:           "v20.12.2"
  JS_RUNTIME_NAME:              "node"
  NODE_PACKAGE_MANAGER:         "bun/1.1.6"

Biome Configuration:
  Status:                       Loaded successfully
  Formatter disabled:           false
  Linter disabled:              false
  Organize imports disabled:    false
  VCS disabled:                 true

Workspace:
  Open Documents:               0
```

### What happened?

Source Location: crates/biome_js_semantic/src/semantic_model/scope.rs:115:33
Thread Name: biome::worker_9
Message: no entry found for key

### Expected result

It should not throw an error.

### Code of Conduct

- [X] I agree to follow Biome's Code of Conduct

**Comment by Sec-ant** [comment]
Duplicate of #2659, there's a [workaround](https://github.com/biomejs/biome/issues/2659#issuecomment-2088284213) in that issue, please do check.
